compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6467,8 +6467,9 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
         }
 
         BType retType = typeParamAnalyzer.getReturnTypeParams(env, bInvokableType.getReturnType());
-        if (restType != symTable.semanticError &&
-                Symbols.isFlagOn(invokableSymbol.flags, Flags.NATIVE) &&
+        long invokableSymbolFlags = invokableSymbol.flags;
+        if (restType != symTable.semanticError && (Symbols.isFlagOn(invokableSymbolFlags, Flags.INTERFACE)
```

Comment:
The `restType` to `retType` break the build. 
Crashes at 
```Unifier::populateParamMapFromRestArg```

```LogTalk
at org.wso2.ballerinalang.compiler.util.Unifier.populateParamMapFromRestArg(Unifier.java:779)
        at org.wso2.ballerinalang.compiler.util.Unifier.createParamMap(Unifier.java:720)
        at org.wso2.ballerinalang.compiler.util.Unifier.build(Unifier.java:94)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkInvocationArgs(TypeChecker.java:6474)

```

Need to properly identify the logic. Just looking at the overall code there is a pattern to make sure that `restType` is not a semantic error type.

Reply:
Ah, just checked the relevant [commit](https://github.com/ballerina-platform/ballerina-lang/commit/cfce4ee3734bb42eb7c92bb5ebe6d448f86ed4fd). I think the current check is correct, my bad.